platform/linux-generic/pktio/loop.c

```diff
@@ -30,6 +30,21 @@
 #define MAX_LOOP 16
 #define LOOP_MTU (64 * 1024)
 
+typedef struct {
+	odp_queue_t loopq;		/**< loopback queue for "loop" device */
+	odp_bool_t promisc;		/**< promiscuous mode state */
+	uint8_t idx;			/**< index of "loop" device */
+} pkt_loop_t;
+
+ODP_STATIC_ASSERT(PKTIO_PRIVATE_SIZE >= sizeof(pkt_loop_t),
+		  "PKTIO_PRIVATE_SIZE too small");
+
+static inline pkt_loop_t *pkt_priv(pktio_entry_t *pktio_entry)
+{
+	return (pkt_loop_t*)(uintptr_t)(pktio_entry->s.pkt_priv);
```

Comment:
This could be made generic and polymorphic via a macro rather than having a version of this for each pktio class. See the `_odp_pkt_get()` macro defined in `packet_inline_types.h` as an example.

Reply:
It would require caller to pass target type to such macro. With my implementation driver just calls `pkt_priv(entry)` and gets access to private data without a need to pass type each time.

Reply:
Agreed.